frontend/src/views/Results.tsx

```diff
@@ -1,10 +1,99 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { useLocation, useHistory } from 'react-router-dom';
+import { Message } from 'stompjs';
+import { LargeText, Text } from '../components/core/Text';
+import { Game, Player, playAgain } from '../api/Game';
+import { checkLocationState } from '../util/Utility';
+import { errorHandler } from '../api/Error';
+import PlayerResultsCard from '../components/card/PlayerResultsCard';
+import { PrimaryButton } from '../components/core/Button';
+import { Room } from '../api/Room';
+import ErrorMessage from '../components/core/Error';
+import Loading from '../components/core/Loading';
+import { disconnect, routes, subscribe } from '../api/Socket';
+import { User } from '../api/User';
+
+const Content = styled.div`
+  padding: 0 20%;
+`;
+
+type LocationState = {
+  game: Game,
+  currentUser: User,
+};
 
 function GameResultsPage() {
+  const history = useHistory();
+  const location = useLocation<LocationState>();
+
+  const [error, setError] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  const [players, setPlayers] = useState<Player[]>();
+  const [currentUser, setCurrentUser] = useState<User | null>(null);
+  const [room, setRoom] = useState<Room | null>(null);
+
+  useEffect(() => {
+    if (checkLocationState(location, 'game', 'currentUser')) {
+      setPlayers(location.state.game.players);
+      setRoom(location.state.game.room);
+      setCurrentUser(location.state.currentUser);
+
+      const subscribeCallback = (result: Message) => {
+        const updatedGame: Game = JSON.parse(result.body);
+
+        // Disconnect users from socket and then redirect them to the lobby page
+        if (updatedGame.playAgain) {
+          disconnect()
+            .then(() => {
+              history.replace(`/game/lobby?room=${updatedGame.room.roomId}`, {
+                user: location.state.currentUser,
+                roomId: updatedGame.room.roomId,
+              });
+            })
+            .catch((err) => setError(err.message));
+        }
+      };
+
+      subscribe(routes(location.state.game.room.roomId).subscribe_game, subscribeCallback)
+        .catch((err) => setError(err.message));
+    } else {
+      history.replace('/game/join', {
+        error: errorHandler('Please join and play a game before viewing the results page.'),
+      });
+    }
+  }, [location, history]);
+
+  const playAgainAction = () => {
+    setError('');
+    setLoading(true);
+
+    playAgain(room!.roomId, { initiator: currentUser! })
+      .catch((err) => {
+        setLoading(false);
+        setError(err.message);
+      });
+  };
```

Comment:
This would require some redesign, but perhaps now or in the future, the results page _is_ the lobby page in a sense? Like maybe it is a modal on top of the lobby page? My reasoning is that if the host unexpectedly leaves, the rest of the players might want to continue the game.

Thinking on this further though, I think the current design is fine for the MVP, but we might want to revamp it in the future.

Reply:
That's a good idea - if it was a modal-type thing on the lobby page, I could see it being pretty simple to change the playAgain function to simply be a onGameEnd function called within the `Game.tsx` file, which redirects you back to the lobby page with like a game/results object. Then on the lobby page it can check for that in `location` and if so display the modal. 

One question about this - if it's a modal with like a close button, what happens if people close it but want to view it again later? Or if they have the modal open and then the host clicks start game, would that catch them off guard (they weren't expecting a new game to begin)?

Reply:
That's a good question -- we should think about design by checking out other sites, since I'm not sure how this would work off the top of my head. We could look at other sites like [colonist.io](http://colonist.io/), [binarysearch.com](https://binarysearch.com/), and [Clash of Code](https://www.codingame.com/multiplayer/clashofcode) for inspiration. One idea is that even if you close the modal, you can re-open it through some tab or something; then, every time you "Play Again", that previous game history is added to the list that you can view on reopening the modal.

We maybe should talk more about this sometime, either on a call or just when playing basketball.

Reply:
Sounds good -  I added all this info to a card in the backlog so I'll resolve this convo for now (since it's probably not ready for this PR yet)